site/_posts/2017-11-30-ray-0.3-release.markdown

```diff
@@ -0,0 +1,147 @@
+[**Ray.tune**][2] --- a new hyperparameter search library, [**distributed actor
+handles**][1], and a number of stability and performance enhancements and bug
+fixes.
+
+To upgrade to the latest version, run
+
+```
+pip install -U ray
+```
+
+## Hyperparameter Search Tool
+
+This release adds [**Ray.tune**][2], a distributed hyperparameter evaluation
+tool for long-running tasks such as reinforcement learning and deep learning
+training. It currently includes the following features:
+
+- Pluggable **early stopping algorithms** including the Median Stopping Rule and
+  [Hyperband][5].
+- Integration with **visualization tools** such as [TensorBoard][6],
+  [rllab's VisKit][7], and a [parallel coordinates visualization][8].
+- Flexible **trial variant generation**, including grid search, random search,
+  and conditional parameter distributions.
+- **Resource-aware scheduling**, including support for concurrent runs of
+  algorithms that require GPUs or are themselves parallel and distributed.
+
+Ray.tune provides a Python API for use with deep learning and other compute
+intensive training tasks. Here is a toy example illustrating usage.
+
+```python
+from ray.tune import register_trainable, grid_search, run_experiments
+
+def my_func(config, reporter):
+    import time, numpy as np
+    i = 0
+    while True:
+        reporter(timesteps_total=i, mean_accuracy=i ** config['alpha'])
+
+        i += config['beta']
+        time.sleep(0.01)
+
+register_trainable('my_func', my_func)
+
+run_experiments({
+    'my_experiment': {
+        'run': 'my_func',
+        'resources': {'cpu': 1, 'gpu': 0},
+        'stop': {'mean_accuracy': 100},
+        'config': {
+            'alpha': grid_search([0.2, 0.4, 0.6]),
+            'beta': grid_search([1, 2]),
+        },
+    }
+})
+```
+
+In-progress results can be visualized live using tools such as Tensorboard and
+rllab's VisKit (or you can read the JSON format logs directly from the driver
+node):
+
+View the [documentation][2] and the [code][9].
+
+### Ray.tune integration with RLlib
+
+You can try out RLlib with Ray.tune with the following example.
+
+```bash
+cd ray/python/ray/rllib
+python train.py -f tuned_examples/cartpole-grid-search-example.yaml
+```
+
+The `tuned_examples` directory also contains pre-tuned hyperparameter
+configurations for common benchmark tasks such as Pong and Humanoid. View the
+[RLlib documentation][3].
+
+### Initial PyTorch support for RLlib
+
+A modern reinforcement learning library should work with multiple deep learning
+frameworks. As a step toward this goal, 0.3 adds support for PyTorch models for
+A3C in RLlib. You can try this out with the following A3C config.
+
+```bash
+cd ray/python/ray/rllib
+./train.py --run=A3C \
+           --env=PongDeterministic-v4 \
+           --config='{"use_pytorch": true, "num_workers": 8, "use_lstm": false, "model": {"grayscale": true, "zero_mean": false, "dim": 80, "channel_major": true}}'
+```
+
+## Distributed Actor Handles
```

Comment:
I don't like this name, but I don't hate it either. Just to capture the channel discussion, my preference would be "Distributed Actor RPC" or "Distributed Actor RMI". Up to you.